src/EmbeddingSpMDM.cc

```diff
@@ -539,6 +540,15 @@ typename ReturnFunctionSignature<inType, indxType, ROWWISE_SPARSE>::
                 block_size * sizeof(uint8_t) + sizeof(float));
             a->vbroadcastss(scale_vreg, scale_src);
             a->vbroadcastss(bias_vreg, bias_src);
+
+            if (pref_dist &&
+                fused_block_size % CACHE_LINE_LEN <= 2 * sizeof(float)) {
+              a->prefetcht0(x86::dword_ptr(
+                  input,
+                  scratchReg2_,
+                  0,
+                  fused_block_size / CACHE_LINE_LEN * CACHE_LINE_LEN));
+            }
```

Comment:
One small concern: if block_size=120 (perhaps this will never happen), this will insert a prefetch at offset 128 which crosses to another row.

Reply:
Sorry about missing your comment. You're absolutely right. Let me create another PR.